File: src/converter/plugins/convertIntersectionType.ts
import ts from "typescript";
import {createSimplePlugin} from "../plugin.js";
import {TypeScriptService, typeScriptServiceKey} from "./TypeScriptPlugin.js";

export const convertIntersectionType = createSimplePlugin((node, context, render) => {
    if (!ts.isIntersectionTypeNode(node)) return null

    const typeScriptService = context.lookupService<TypeScriptService>(typeScriptServiceKey)

    return `Any /* ${typeScriptService?.printNode(node)} */`;
})

File: karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/extension/plugins/convertIntersectionType.kt
package io.github.sgrishchenko.karakum.extension.plugins

import io.github.sgrishchenko.karakum.extension.createPlugin
import typescript.isIntersectionTypeNode

val convertIntersectionType = createPlugin plugin@{ node, context, _ ->
    if (!isIntersectionTypeNode(node)) return@plugin null

    val typeScriptService = context.lookupService(typeScriptServiceKey)

    "Any /* ${typeScriptService?.printNode(node)} */"
}
